doc/rst/technotes/dsi.rst

```diff
@@ -954,5 +954,173 @@ implicitly for each ``GlobalDomain`` object and so will be local in any case.
+
+Known vs. Any methods
+---------------------
+
+There are two kinds of transfers accounted for in this interface.
+
+The first kind of transfer is a ``Known`` transfer. These are transfers that
+the implementer believes to be optimal. Typically this means that the
+implementer knows the type and internals of the other domain map involved in
+the transfer.
+
+The second kind of transfer is an ``Any`` transfer. These are transfers to/from
+domain maps whose type and internals are **not** known. These transfers rely on
+a common interface, which may involve unnecessary overhead compared to a
+``Known`` transfer. For example, it is likely that many domain maps will
+implement ``Known`` transfers to/from Chapel's default rectangular arrays. It
+is also likely that many domain maps will be implemented using default
+rectangular arrays. Knowing this, implementers of ``Any`` transfers can
+implement in terms of ``Known`` transfers to/from their own default rectangular
+arrays as a common interface.
+
+This interface recognizes these kinds of transfers because otherwise it is
+difficult to determine which distribution implements the best transfer.
+
+For example, the author of ``PopularDist`` could implement ``Any`` methods for
+better performance with unknown third-party domain maps. The author of
+``NewDist`` could implement ``Known`` methods to ``PopularDist`` due to its
+popularity. By doing so the author of ``NewDist`` has informed Chapel that it
+knows better than ``PopularDist`` and a more optimal transfer occurs.
+
+Chapel will call ``Known`` methods before ``Any`` methods if they can be
+resolved. A recommended convention is to implement ``Known`` methods with a
+where-clause that constrains the type of the ``otherClass`` argument. This
+prevents the ``Known`` method from being called with unknown types.
+
+When Does a Bulk-Transfer Occur?
+--------------------------------
+
+There are a number of conditions that need to be true for a bulk-transfer to
+occur:
+
+* The config-param ``useBulkTransfer`` must be true
+
+* The element types of the arrays must be identical
+
+* The element type must be one of the following:
+    * ``integral``
+    * ``real``
+    * ``complex``
+    * record for which ``isPODType`` returns true
+    * tuple for which ``isPODType`` returns true
+
+The element type is restricted in order to preserve copy semantics of each
+element. For example, ``integral`` types can be bit-copied and so have no
+side-effects. But a record with a copy-initializer may have side effects that
+should not be eliminated by the bulk-transfer optimization.
+
+Transfers for Aliasing Domain Maps
+----------------------------------
+
+It is valid to write a domain map that aliases another domain map's data. For
+example, a user could write their own variant of a Chapel array slice in terms
+of the Domain Map Standard Interface.
+
+When aliasing other domain maps it is recommended to only implement ``Known``
```

Comment:
It'd be friendly for references to "Known" and "Any" outside of their section to link back to the section in which they were defined.